ui_framework/src/components/expression/expression_item_popover.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { KuiOutsideClickDetector } from '../outside_click_detector';
+
+const POPOVER_ALIGN = [
+  'left',
+  'right',
+];
+
+const KuiExpressionItemPopover = ({
+  className,
+  title,
+  isVisible,
+  children,
+  align,
+  onOutsideClick,
+  ...rest
+}) => {
+  const classes = classNames('kuiExpressionItem__popover', className, {
+    'kuiExpressionItem__popover--isHidden': !isVisible,
+    'kuiExpressionItem__popover--alignRight': align === 'right'
+  });
+  return (
+    <KuiOutsideClickDetector onOutsideClick={onOutsideClick}>
+      <div
+        className={classes}
+        {...rest}
+      >
+        <div className="kuiExpressionItem__popoverTitle">
+          {title}
+        </div>
+        <div className="kuiExpressionItem__popoverContent">
+          {children}
+        </div>
+      </div>
+    </KuiOutsideClickDetector>
+  );
+};
+
+KuiExpressionItemPopover.defaultProps = {
+  align: 'left',
+};
+
+KuiExpressionItemPopover.propTypes = {
+  className: PropTypes.string,
+  title: PropTypes.string.isRequired,
+  isVisible: PropTypes.bool.isRequired,
+  children: PropTypes.node,
+  align: PropTypes.oneOf(POPOVER_ALIGN),
+  onOutsideClick: PropTypes.func.isRequired,
+};
```

Comment:
What do you think about not requiring `isVisible` and `onOutsideClick` and providing `defaultProps` for them (and `align: 'left'`)? That would make the tests a bit less verbose.

Reply:
@weltenwort Thank you for that!
`isVisible` is not needed anymore because `KuiExpressionItemPopover`
will be added to `KuiExpressionItem` only if the popover is needed.
I remove `isVisible` and `.kuiExpressionItem__popover--isHidden`
and adapt the tests.

I assume the default usage of `KuiExpressionItemPopover`
is to vanish for an outside click and to accomplish that the client must be
notified about the outside click, so as I see it's better to let `onOutsideClick` explicit.

`align: 'left' `has already a default prop.